### PR TITLE
[CPD-1925]: add option to choose school latest LP/DP combo when adding a participant

### PR DESCRIPTION
**Pull request by ltello**
### Context

Allow SITs to choose the LP/DP combo of their school's latest cohort when adding a new participant.

image
[Ticket](https://dfedigital.atlassian.net/browse/CST-1925) 

### Changes proposed in this pull request
- Redefine the logic for the SIT to confirm the current participant's LP/DP combo
- Redefine the logic for the SIT to choose school programme
- Display new option with the extra combo when applicable

### CODE REVIEW

File: app/forms/schools/add_participants/wizard_steps/choose_mentor_step.rb [hunk 1]
```diff
@@ -15,9 +15,12 @@ def self.permitted_params
         end
 
         def next_step
+          # debugger
```
Comment (tonyheadford) [inline app/forms/schools/add_participants/wizard_steps/choose_mentor_step.rb:18]:
```suggestion
```

File: app/forms/schools/add_participants/wizard_steps/continue_current_programme_step.rb [hunk 2]
```diff
@@ -15,7 +15,8 @@ def self.permitted_params
         end
 
         def next_step
-          if continue_current_programme?
+          # debugger
```
Comment (tonyheadford) [inline app/forms/schools/add_participants/wizard_steps/continue_current_programme_step.rb:18]:
```suggestion
```

File: app/forms/schools/add_participants/wizard_steps/email_step.rb [hunk 3]
```diff
@@ -15,6 +15,7 @@ def self.permitted_params
         end
 
         def next_step
+          # debugger
```
Comment (tonyheadford) [inline app/forms/schools/add_participants/wizard_steps/email_step.rb:18]:
```suggestion
```

**Review by tonyheadford — CHANGES_REQUESTED** [review]
I think the code is fine just a few commented out bits to remove

File: app/views/schools/add_participants/_transferring_ect_or_mentor_answers.html.erb [hunk 4]
```diff
@@ -78,31 +78,40 @@
...
-                    visually_hidden_text: "mentor",
-                    href: form.change_path_for(step: :"appropriate-body")) %>
+                         visually_hidden_text: "mentor",
+                         href: form.change_path_for(step: :"appropriate-body")) %>
     <% end %>
   <% end %>
 
   <% if form.show_training_provider_section? %>
     <% summary_list.with_row do |row| %>
+      <% # debugger %>
```
Comment (frankieroberto) [inline app/views/schools/add_participants/_transferring_ect_or_mentor_answers.html.erb:88]:
```suggestion
```

File: app/helpers/application_helper.rb [hunk 5]
```diff
@@ -79,6 +79,13 @@ def possessive_name(name)
     "#{name}#{name[-1] == 's' ? '’' : '’s'}"
   end
 
+  def keep_providers_options
+    [
+      OpenStruct.new(id: "yes", name: "Yes"),
+      OpenStruct.new(id: "no", name: "No, they'll move to different training providers or a different training programme"),
```
Comment (frankieroberto) [inline app/helpers/application_helper.rb:85]: Curly apostrophe.

```suggestion
      OpenStruct.new(id: "no", name: "No, they’ll move to different training providers or a different training programme"),
```

File: app/forms/schools/add_participants/transfer_wizard.rb [hunk 6]
```diff
@@ -135,23 +135,26 @@ def withdrawn_participant?
...
         )
       end
 
-      def transfer_fip_participant_and_continue_existing_programme(profile)
-        Induction::TransferAndContinueExistingFip.call(
-          school_cohort:,
+      def transfer_fip_participant_to_schools_programme(profile)
+        chosen_school_cohort = join_current_cohort_school_programme? ? school_current_cohort : school_cohort
+
+        # debugger
```
Comment (frankieroberto) [inline app/forms/schools/add_participants/transfer_wizard.rb:152]:
```suggestion
```

**Review by frankieroberto — APPROVED** [review]

**Pull request merged**